42/README.md

```diff
@@ -70,14 +71,133 @@ The above process ensure the node keeps validating and accepting new blocks whil
+```
+
+Validators can watch how the election goes:
+
+```
+$ bigchaindb migration status <migration-id>
+```
+
+Outputs:
+```
+votes_recieved=<Sum_of_votes_recieved>
+votes_allocated=<Sum_of_votes_allocated_in_election>
+network_size=<Total_network_power>
+election_status=<Election_status>
+```
+
+#### 2. Replay the blockchain
+
+In order to replay the chain up to the migration height, we need to either keep the old Tendermint version running or skip replaying the corresponding part of the chain = archive the chain. We consider the former to be a huge hassle so we describe an archiving scheme further in this BEP.
+
+#### 3. Start the new chain
+
+Validators have to install and launch the new version of Tendermint. They need to prepare a new `genesis.json`. The new genesis file has to contain the validator set, the application hash (both at the migration height), and the identifier of the new Tendermint chain (`chain_id`).
+
+The application hash is calculated by computing the Merkle tree root of the current UTXO. Note that hash calculation is a subject to design in a separate BEP and can be changed in the future.
+
+`chain_id` is generated and stored by BigchainDB upon conclusion of the migration election. When Tendermint sends this ID as part of the `InitChain` ABCI request, BigchainDB understands that the user has switched to a new Tendermint version so BigchainDB switches to accept new transactions and blocks.
+
+To offer a convenient way to get the data, we propose a CLI command:
+
+```
+$ bigchaindb status
+```
+
+Its output can be extended over time to solve different purposes. For a start, it has to contain the latest known list of validators, the latest known app hash, and the identifier of the Tendermint chain.
+```
+{
+    "chain_id": "...",
+    "validators": [{
+        "pub_key": {
+            "type": "...",
+            "value": "..."
+        },
+        "power": "...",
+        "name": "..."
+    },
+    "app_hash": "..."
+}
+```
+
+Validators upgrading Tendermint are supposed to execute the command above and copy the validators, the app hash, and the chain ID into their `genesis.json`.
+
+Validators joining the network after the migration take the genesis file from existing validators, as usual.
+
+#### 4. Join the network after a migration
+
+When a validator joins the network after a migration, it does not suffice for him to know `genesis.json` - he also needs the archive of the old chain.
+There might be more than one migration, so the initial chain might go from height 0 to 33277, the second chain might go to height 88234, the third chain can go from 88234 up to the recent height. In this case the validator has to get an archive containing archived blocks from height 0 to height 88234.
+
+Such a validator needs to take a snapshot of the old chain from an existing member of the network. The snapshot can be created via a CLI command:
```

Comment:
In the first iteration of this implementation, we can just recommend the user to use `mongodump` and `mongorestore` for achiving and loading data.

Reply:
Yes, we need to be explicit about the importance of keeping/publishing the dump together with `genesis.json`.

Reply:
I think Tendermint stores more information than MongoDB stores, such as all the block headers. Shouldn't that be "dumped" as well?

Reply:
After a migration, Tendermint starts from scratch so no - there is nothing to dump.